### PR TITLE
[incubator/vault-operator] Introducing the Vault Operator chart

### PR DESCRIPTION
**Pull request by mlaccetti**
**What this PR does / why we need it**:

Add in a chart for the `vault-operator`

**Which issue this PR fixes**:

Fix as PR

**Special notes for your reviewer**:

Based off of the `etcd-operator`

### CODE REVIEW

**Comment by mlaccetti** [comment]
@unguiculus any chance I can get your 👀 on this? 😄

**Comment by paddie** [comment]
An official helm chart for this would be very welcome indeed!

File: incubator/vault-operator/values.yaml [hunk 1]
```diff
@@ -0,0 +1,71 @@
...
+  failureThreshold: 3
+
+###
+# All of the config variables related to setting up the etcd-operator
+# Disabled by default
+# If you want more information about the variables exposed, please visit:
+#   https://github.com/kubernetes/charts/tree/master/stable/etcd-operator#configuration
+###
+etcd-operator:
+  enabled: false
```
Comment (jody-frankowski) [inline incubator/vault-operator/values.yaml]: Wouldn't it be better if this was enabled by default?

Comment (mlaccetti) [reply incubator/vault-operator/values.yaml]: I'm not entirely sure - we run `etcd-operator` as a global, so one and done; I think it should be left up to the person using the `vault-operator` to be able to choose if they want 1:1 or 1:n.

**Review by unguiculus — CHANGES_REQUESTED** [review]
Please compare with what you get from `helm create` using Helm 2.8+ and update accordingly.

**Comment by ccojocar** [comment]
@mlaccetti @unguiculus @jody-frankowski Any chance to get this merged soon? I would be willing to put some effort on my side to help out. Would you have time to sort out the open issues?

Thanks

cc @rawlingsj @jstrachan

**Comment by mlaccetti** [comment]
@ccojocar Ah, just got lost in the shuffle - just need to do the `helm-2.7` to `helm-2.9` chart tweaks. I'll get this done today.

**Comment by ccojocar** [comment]
@mlaccetti Great! Thanks

**Comment by mlaccetti** [comment]
@unguiculus I've updated to match (as much as possible) the Helm 2.8+ chart style.

**Comment by ccojocar** [comment]
@unguiculus Please could you have a look at the latest changes? Are there any open issues? I tested  it myself and it worked fine. It would be great if you could merge it.

cc @scottrigby @tompizmor @viglesiasce @mattfarina

File: incubator/vault-operator/values.yaml [hunk 2]
```diff
@@ -0,0 +1,80 @@
...
+rbac:
+  create: true
+  apiVersion: v1beta1
+
+## Service account names and whether to create them
+serviceAccount:
+  create: true
+  name: vault-operator-sa
+
+resources:
```
Comment (unguiculus) [inline incubator/vault-operator/values.yaml]: Don't specify defaults for resources. Comment them out.

File: incubator/vault-operator/values.yaml [hunk 3]
```diff
@@ -0,0 +1,80 @@
...
+
+## Install Default RBAC roles and bindings
+rbac:
+  create: true
+  apiVersion: v1beta1
+
+## Service account names and whether to create them
+serviceAccount:
+  create: true
+  name: vault-operator-sa
```
Comment (unguiculus) [inline incubator/vault-operator/values.yaml]: Leave default empty

File: incubator/vault-operator/values.yaml [hunk 4]
```diff
@@ -0,0 +1,80 @@
...
+replicaCount: 1
+image:
+  repository: quay.io/coreos/vault-operator
+  tag: 0.1.9
+  pullPolicy: Always
+
+## Install Default RBAC roles and bindings
+rbac:
+  create: true
+  apiVersion: v1beta1
```
Comment (unguiculus) [inline incubator/vault-operator/values.yaml]: Used `v1`. Don't make it configurable.

File: incubator/vault-operator/values.yaml [hunk 5]
```diff
@@ -0,0 +1,80 @@
+## Default values for the image
+name: vault-operator
+replicaCount: 1
+image:
+  repository: quay.io/coreos/vault-operator
+  tag: 0.1.9
+  pullPolicy: Always
```
Comment (unguiculus) [inline incubator/vault-operator/values.yaml]: IfNotPresent

File: incubator/vault-operator/README.md [hunk 6]
```diff
@@ -0,0 +1,141 @@
...
+
+This chart bootstraps a vault-operator and allows the deployment of vault cluster(s). It depends on the `etcd-operator` being installed.
+
+## Official Documentation
+
+Official project documentation found [here](https://github.com/coreos/vault-operator)
+
+## Pre-requisites
+
+- Kubernetes 1.8+
```
Comment (unguiculus) [inline incubator/vault-operator/README.md]: You may want to go for 1.9+ and use `apps/v1` for the deployment.

**Review by unguiculus — CHANGES_REQUESTED** [review]

File: incubator/vault-operator/values.yaml (hunk 2 above)
Comment (mlaccetti) [reply incubator/vault-operator/values.yaml]: This is an operator, so it made sense to set some defaults, but I can comment them and put in an empty default.

File: incubator/vault-operator/README.md (hunk 6 above)
Comment (mlaccetti) [reply incubator/vault-operator/README.md]: We're stuck on 1.8, hence the `v1beta2`, but everybody else shouldn't suffer, so I'll bump.

**Review by unguiculus — APPROVED** [review]

**Pull request merged**